code/datums/status_effects/status_effect.dm

```diff
@@ -0,0 +1,243 @@
+//Status effects are used to apply temporary or permanent effects to mobs. Mobs are aware of their status effects at all times.
+//This file contains their code, plus code for applying and removing them.
+//When making a new status effect, add a define to status_effects.dm in __DEFINES for ease of use!
+
+/datum/status_effect
+	var/id = "effect" //Used for screen alerts.
+	var/duration = -1 //How long the status effect lasts in DECISECONDS. Enter -1 for an effect that never ends unless removed through some means.
+	var/tick_interval = 10 //How many deciseconds between ticks, approximately. Leave at 10 for every second.
+	var/mob/living/owner //The mob affected by the status effect.
+	var/status_type = STATUS_EFFECT_UNIQUE //How many of the effect can be on one mob, and what happens when you try to add another
+	var/on_remove_on_mob_delete = FALSE //if we call on_remove() when the mob is deleted
+	var/examine_text //If defined, this text will appear when the mob is examined - to use he, she etc. use "SUBJECTPRONOUN" and replace it in the examines themselves
+	var/alert_type = /obj/screen/alert/status_effect //the alert thrown by the status effect, contains name and description
+	var/obj/screen/alert/status_effect/linked_alert = null //the alert itself, if it exists
+
+/datum/status_effect/New(list/arguments)
+	on_creation(arglist(arguments))
+
+/datum/status_effect/proc/on_creation(mob/living/new_owner, ...)
+	if(new_owner)
+		owner = new_owner
+	if(owner)
+		LAZYADD(owner.status_effects, src)
+	if(!owner || !on_apply())
+		qdel(src)
+		return
```

Comment:
```suggestion
		CRASH("Status effect [type] created improperly. Owner: [isnull(owner) "null" : "[owner]"].")
```

Reply:
This is from tg, if they dont have this issue its highly likely we wont either

Reply:
The suggestion is just making a silent error not be silent. It's free, and if it's never triggered, that is fine, but then the check shouldn't be there in the first place.

Reply:
I'm reasonable certain theres cases where it happens as a normal part of status effects and shouldnt error